### DISCUSSION TITLE
[master]小米路由R4A千兆版WIFI无法连接

### DISCUSSION THREAD

**Discussion by chobitsos** [immortalwrt/immortalwrt#533] [General]
编译正常
2G 5G双信号正常
连接上去之后立马断开
偶尔出现密码错误提示
日志显示
daemon.info hostapd: wlan1: STA 4a:b3:8c:eb:bf:ed IEEE 802.11: authenticated
daemon.info hostapd: wlan1: STA 4a:b3:8c:eb:bf:ed IEEE 802.11: associated (aid 1)
daemon.info hostapd: wlan0: STA 4a:b3:8c:eb:bf:ed IEEE 802.11: authenticated
daemon.info hostapd: wlan0: STA 4a:b3:8c:eb:bf:ed IEEE 802.11: associated (aid 1)
daemon.notice hostapd: wlan1: Prune association for 4a:b3:8c:eb:bf:ed
daemon.info hostapd: wlan0: STA 4a:b3:8c:eb:bf:ed IEEE 802.11: deauthenticated due to local deauth request

尝试解决一
更换的Makefile前几个版本文件，编译成功 同样错误无法连接

尝试解决二
wireless增加
```
option disassoc_low_ack '0'
option wpa_strict_rekey '1'
option wpa_group_rekey '86400'
```
同样错误，无法连接

尝试解决三
取消开源驱动，更换问闭源驱动，编译成功
双信号正常 连接正常，一切正常，就是闭源驱动太难用了

然后还有各种尝试开源驱动始终无法使用
下了隔壁lean大神的源码，连接正常信号正常，一切正常
下了隔壁x-wrt大神的源码，连接正常信号正常，一切正常
同样内核，同样配置，到底哪里出了问题？？？？

**Comment by 1715173329** [comment]
as you're at snapshot, it's normal.
try openwrt-21.02 branch.

**Reply by chobitsos** [reply]
> as you're at snapshot, it's normal. try openwrt-21.02 branch.

可以修一下吗大神，openwrt-21.02分支好多机器都没适配啊，没YOUKU L1C

**Reply by 1715173329** [reply]
> 可以修一下吗大神

It's a upstream issue, that's how wireless driver works, I cannot address it.

> 没YOUKU L1C

Move to openwrt-18.06-k5.4 then.

**Reply by chobitsos** [reply]
😂好像是个坑，有任何修复的思路或者头绪吗，我可以自己尝试修一修，即使时上游问题也可以尝试一下，略懂一些嵌入式开发和代码

**Reply by chobitsos** [reply]
放弃了，我在试试用openwrt-21.02 branch适配个YOUKU-LC1，这不就香了吗

**Reply by 1715173329** [reply]
> 放弃了，我在试试用openwrt-21.02 branch适配个YOUKU-LC1，这不就香了吗

直接 cherry-pick 4a9f389ed2dcee18a5c5e1f0d4e5c406f9290579 到 21.02 即可。

**Reply by chobitsos** [reply]
> > 放弃了，我在试试用openwrt-21.02 branch适配个YOUKU-LC1，这不就香了吗
> 
> 直接 cherry-pick [4a9f389](https://github.com/immortalwrt/immortalwrt/commit/4a9f389ed2dcee18a5c5e1f0d4e5c406f9290579) 到 21.02 即可。

卧槽openwrt-21.02 branch的WIFI也不行，R4A也连不上，😂

**Reply by chobitsos** [reply]
难道我只能用闭源驱动了吗😂😂😂😂

**Reply by 1715173329** [reply]
你可以试试官方 snapshot 的固件，如果还有问题那么就没办法了
https://downloads.openwrt.org/snapshots/targets/ramips/mt7620/openwrt-ramips-mt7620-youku_yk-l1c-squashfs-sysupgrade.bin

听说新的 mt76 对于老的 mt 无线支持并不太好，我没有实际测过（很久没用 ramips 了
目前 openwrt-18.06 分支的无线出了问题，还在排查问题

**Reply by chobitsos** [reply]
> 你可以试试官方 snapshot 的固件，如果还有问题那么就没办法了 https://downloads.openwrt.org/snapshots/targets/ramips/mt7620/openwrt-ramips-mt7620-youku_yk-l1c-squashfs-sysupgrade.bin
> 
> 听说新的 mt76 对于老的 mt 无线支持并不太好，我没有实际测过（很久没用 ramips 了 目前 openwrt-18.06 分支的无线出了问题，还在排查问题

这是YOUKU的，YOUKU是正常的，小米的R4A不行，官方的固件没有改DTS不能直刷，我去拉个源码试试

**Reply by chobitsos** [reply]
> 你可以试试官方 snapshot 的固件，如果还有问题那么就没办法了 https://downloads.openwrt.org/snapshots/targets/ramips/mt7620/openwrt-ramips-mt7620-youku_yk-l1c-squashfs-sysupgrade.bin
> 
> 听说新的 mt76 对于老的 mt 无线支持并不太好，我没有实际测过（很久没用 ramips 了 目前 openwrt-18.06 分支的无线出了问题，还在排查问题

拉了官方源码，编译了小米4A千兆版MT76开源固件，一切正常，可能要同步一下上游，我可以寄一个小米4A路由器给你

**Reply by 1715173329** [reply]
刚合并了上游 master 源码，请重新生成 buildroot config。

**Reply by chobitsos** [reply]
我佛了，编译出来后WIFI不起来了
ifconfig 没有看到无线接口
```
#brctl show
bridge name     bridge id               STP enabled     interfaces
br-lan          7fff.50d2f5f3cf33       no              lan2
                                                        lan1
```

#lsmod | grep mt 驱动都已经加载
```
cfg80211              293011  5 mt76x2_common,mt76x02_lib,mt7603e,mt76,mac80211
mac80211              556776  4 mt76x2e,mt76x02_lib,mt7603e,mt76
mt76                   46528  4 mt76x2e,mt76x2_common,mt76x02_lib,mt7603e
mt7603e                40348  0
mt76x02_lib            44108  2 mt76x2e,mt76x2_common
mt76x2_common          12546  1 mt76x2e
mt76x2e                10721  0
```
合并之后内核直接到5.10了😂
附上log给你看看，要硬件的话我可以发台给你
[syslog.log](https://github.com/immortalwrt/immortalwrt/files/7735039/syslog.log)
[Kernel.log](https://github.com/immortalwrt/immortalwrt/files/7735045/Kernel.log)
[config编译配置.txt](https://github.com/immortalwrt/immortalwrt/files/7735053/config.txt)

**Reply by 1715173329** [reply]
我先拿我的新三试试

**Reply by 1715173329** [reply]
这是我的配置，新三测试倒是可以正常启动、正常连接、正常通讯。
最精简配置，仅替换 wpad 为 openssl。
```buildroot
CONFIG_TARGET_ramips=y
CONFIG_TARGET_ramips_mt7621=y
CONFIG_TARGET_ramips_mt7621_DEVICE_d-team_newifi-d2=y
# CONFIG_PACKAGE_libwolfssl is not set
# CONFIG_PACKAGE_wpad-basic-wolfssl is not set
CONFIG_PACKAGE_wpad-openssl=y
# CONFIG_WPA_WOLFSSL is not set
```

**Reply by chobitsos** [reply]
> 这是我的配置，新三测试倒是可以正常启动、正常连接、正常通讯，不过速率有点一言难尽。
> 
> ```
> CONFIG_TARGET_ramips=y
> CONFIG_TARGET_ramips_mt7621=y
> CONFIG_TARGET_ramips_mt7621_DEVICE_d-team_newifi-d2=y
> # CONFIG_PACKAGE_libwolfssl is not set
> # CONFIG_PACKAGE_wpad-basic-wolfssl is not set
> CONFIG_PACKAGE_wpad-openssl=y
> # CONFIG_WPA_WOLFSSL is not set
> ```

速率倒是无所谓，那我试试make clean一下然后在rm-rf ./tmp && rm-rf ./.config,重新编译一下

**Comment by chobitsos** [comment]
已出固件，依然没有起来，换了openssl，我想放弃了，实在不行只能用闭源驱动了
```
# lsmod | grep mt
cfg80211              293011  5 mt76x2_common,mt76x02_lib,mt7603e,mt76,mac80211
mac80211              556776  4 mt76x2e,mt76x02_lib,mt7603e,mt76
mt76                   46528  4 mt76x2e,mt76x2_common,mt76x02_lib,mt7603e
mt7603e                40348  0
mt76x02_lib            44108  2 mt76x2e,mt76x2_common
mt76x2_common          12546  1 mt76x2e
mt76x2e                10721  0
```
mt7603e看样子是7603起不来，不知道为啥

```
# brctl show
bridge name     bridge id               STP enabled     interfaces
br-lan          7fff.50d2f5f3cf33       no              lan2  lan1
```

log里面压根就好像没有加载驱动
[syslog.log](https://github.com/immortalwrt/immortalwrt/files/7737206/syslog.log)
[Kernel.log](https://github.com/immortalwrt/immortalwrt/files/7737207/Kernel.log)
[config编译配置换openssl.txt](https://github.com/immortalwrt/immortalwrt/files/7737208/config.txt)

**Comment by chobitsos** [comment]
大型社死现场，刚刚编译了开源驱动,一切正常，全部正常
究其原因，我他妈的放了两个路由器在房间，两个路由器刷了同一个EEPROM
导致两个路由器的BSSID的MAC地址一样，卧槽了，卧槽了，就因为这事情折腾了三四个通宵
其实源码毛线问题都没有，出问题的是我自己，真的浪费了大家的时间，对不起了

怎么发现这个问题的，我在编译了闭源驱动后发现也连不上WIFI，跟开源驱动一样，想了半天没想到，然后无意中重启掉了一台路由器，在这个空挡的时间WIFI立马连上了，然后就开始怀疑，ifconfig一下，果然两台路由器两个WIFI接口的MAC地址都是一样的，突然不自信了，我悟了，我悟了！
